packages/trip-form/src/TripOptions/FeaturedOptionOverlay.tsx

```diff
@@ -0,0 +1,71 @@
+import { ExternalLinkAlt } from "@styled-icons/fa-solid/ExternalLinkAlt";
+import React, { FunctionComponent } from "react";
+
+import * as S from "./styled";
+import { Company, Modes } from "./types";
+import { getCategoryModes, getCategoryPrimaryMode } from "./util";
+
+const FeaturedOptionOverlay = ({
+  featuredOption,
+  setFeaturedOption,
+  supportedModes,
+  CompanyIcon,
+  DetailedModeIcon
+}: {
+  featuredOption: string;
+  setFeaturedOption(option: string): void;
+  supportedCompanies: Company[];
+  supportedModes: Modes;
+  CompanyIcon?: FunctionComponent<{ company: string }>;
+  DetailedModeIcon?: FunctionComponent<{ mode: string }>;
+}): JSX.Element => {
+  // Find the mode that matches the selected category
+  const category = supportedModes.categories.find(
+    c =>
+      // Each supported mode may have "sub-modes". We need to identify the correct one to match the category correctly
+      featuredOption ===
+      getCategoryModes(c).find(mode => featuredOption === mode)
+  );
+
+  const defaultImageRender = o =>
+    o.image ? <img src={o.image} alt={o.label} /> : o.label;
+
+  return (
+    <S.OverlayContainer>
+      <button onClick={() => setFeaturedOption(null)} type="button">
+        Back
+      </button>
+      <S.FeaturedOptionImageWrapper>
+        {(DetailedModeIcon && (
+          <DetailedModeIcon mode={getCategoryPrimaryMode(category)} />
+        )) || <S.MaxHeightImage src={category.image} />}
+      </S.FeaturedOptionImageWrapper>
+      <S.OverlayHeader>{category.label}</S.OverlayHeader>
+      <p>{category.description}</p>
+      <S.OverlayOptions>
+        {category.options.map(o => {
+          // Don't show non-company options (e.g., park and ride)
+          if (!o.company) return null;
+          return (
+            <li key={o.label}>
+              <a href={o.url}>
+                <span className="label">
+                  {(CompanyIcon && <CompanyIcon company={o.company} />) ||
+                    defaultImageRender(o)}
```

Comment:
When a custom icon is passed in via `CompanyIcon` its parent span is ending up with a height of 0 and thus so does its icon component child.  The icon height should be equivalent to height of the list item minus some padding which I think can be achieved by setting an explicit height on the aforementioned span

Would be nice if you could setup story that uses all of these custom icon props so you can get a better idea of how they're functioning

Reply:
Here's the mock of the scooter icons to give you an idea of what size they should be:

![scooter-icons](https://user-images.githubusercontent.com/1002875/135510368-4ce1950e-de72-4567-83a2-53ec0fe6527b.PNG)